src/python/pants/reporting/streaming_workunit_handler.py

```diff
@@ -3,11 +3,25 @@
 
 import threading
 from contextlib import contextmanager
-from typing import Any, Callable, Iterable, Iterator, Optional
+from dataclasses import dataclass
+from typing import Any, Callable, Iterable, Iterator, List, Optional
 
+from pants.engine.fs import Digest
+from pants.engine.internals.scheduler import SchedulerSession
 from pants.util.logging import LogLevel
 
 
+@dataclass(frozen=True)
+class StreamingWorkunitContext:
+    _scheduler: SchedulerSession
+
+    def digests_to_bytes(self, digests: List[Digest]) -> List[bytes]:
```

Comment:
Should be able to accept sequence, not just a list, also return a tuple.
```  def digests_to_bytes(self, digests: Sequence[Digest]) -> Tuple[bytes, ...]:```

Reply:
Is a sequence the right type constraint? The order of bytes objects in the output maps to the order of Digest objects in the input, so order matters and I think that `Sequence` involves inherently unordered types like sets? Similarly, we care about `Tuple` rather than `List` in the engine since `Tuple`s are immutable, but in this case I'm not sure if it's necessary to use a Tuple instead of a List.

Reply:
I think sequence is actually ordered (as opposed to collection) for example `Set[str]` is not a valid `Sequence[str]`

@Eric-Arellano will know more.